example/lib/samples/complex_sample.dart

```diff
@@ -33,6 +33,12 @@ class ComplexSample extends StatelessWidget {
         'time': TimeOfDay.now(),
         'booleanObject':
             FormControl<BooleanObject>(value: BooleanObject('Yes')),
+        'position': EmbeddedFormControl<MapEntry<double, double>>(
+          toEmbedded: (value) => {
+            'lat': value?.key,
+            'lng': value?.value,
+          },
+        ),
       }, [
```

Comment:
Take this `position` as an example.
When using FormGroup, the output `form.value` is `{..., time: xxx, position: {lat: 0.0, lng: 0.0}}`
When using EmbeddedFormControl, the output of `form.value` is `{..., time: xxx, lat: 0.0, lng: 0.0}`

Many times the server needs the data in the second format. If you use FormGroup, you also need to convert the data format manually.

Reply:
Hi @ipcjs, thanks for your interest and contribution. I agree with @vasilich6107 that not always is a good idea to pass data directly from Form to Server.

Another interesting thing is that we don' t have to much details about your use case, but I'm just wondering myself:

If you want the data in the format: `{..., time: xxx, lat: 0.0, lng: 0.0}` why not just create a FormGroup with that same format instead of creating a more complex nested FormGroup? If you need your data to be flat, then why not just create it in a flat format and that's it?

Don't make me wrong, I think you have brought up an interesting point, but sometimes solutions are simpler than we thought. Instead of adding more code to maintain, we can just make it simpler.

Reply:
Another built-in solution is to bind two different Reactive Widgets to the same FormControl. But this FormControl is an object, let's say `Coordinates`.

` ` `dart
class Coordinates {
final double lat;
final double long;
}
` ` `

and Widget A and Widget B are bound to the same Coordinate. But with the help of a `ControlValueAccessor`, you can map a Widget with a property of the Object.

Reply:
In most cases it is possible to pass data directly from the Form to Server. In the remaining few cases, it cannot be passed directly simply because of inconsistencies in the format of individual fields. That's why I wrote EmbeddedFormControl😅

Reply:
1-) What if this EmbeededFormControl overrides one of the parent (FormGroup) control? For example this case:

```
FormGroup({
   'name': .....,
   'friend': EmbeddedFormControl({name:..., surname:....})
})
```

2-) What will happen if I put an `EmbeddedFormControl` as a child of a `FormArray`?

3-) You have changed the implementation of the `get FormGroup.value` in order to add the values of the `EmbeddedFormControl`. But what about the implementation of the `set FormGroup.value`? Is there going to be an inconsistency in the way we `get` and `set` the value to a FormGroup?

To many questions that I don't find an easy answare for them.

Reply:
It is indeed confusing to modify the return value of `form.value` directly through `EmbeddedFormControl`.

What we need is actually a method like `toJson` to convert the `FormControl` value. 🤔️